### PR TITLE
Fix avatar recording playback mouth not moving

### PR DESCRIPTION
**Pull request by ctrlaltdavid**

### CODE REVIEW

**Comment by ctrlaltdavid** [comment]
Testing

	Setup:
	- Two testers running PR Interface on server with PR Sandbox.
	- Each tester has avatar with mouth supporting blendshapes (e.g., Being of Light).
	- Recording script from PR running on Interface: record.js.
	- Recording playback script from PR running on Sandbox: playRecordingAC.js with 1 or more instances.

	Tests:
	- When you speak you see your own avatar's mouth move (e.g., in full screen mirror).
	- When the other person speaks you see their avatar's mouth move.
	- Make a recording while speaking such that your avatar's mouth moves. When the recording plays back the playback avatar's mouth moves.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 98b943269bb49116feded6f3efe313b1f70a460e available at: https://gist.github.com/96f777337db3e3b4cf2e78cfb08e044f<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10443-98b9432.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10443-98b9432.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: e426d70437bc214263ae0e53992da125398dc668 available at: https://gist.github.com/96f777337db3e3b4cf2e78cfb08e044f<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10443-e426d70.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10443-e426d70.dmg

File: libraries/avatars/src/HeadData.cpp [hunk 1]
```diff
@@ -163,8 +166,8 @@ void HeadData::fromJson(const QJsonObject& json) {
             QJsonArray blendshapeCoefficientsJson = jsonValue.toArray();
             for (const auto& blendshapeCoefficient : blendshapeCoefficientsJson) {
                 blendshapeCoefficients.push_back((float)blendshapeCoefficient.toDouble());
-                setBlendshapeCoefficients(blendshapeCoefficients);
             }
+            setBlendshapeCoefficients(blendshapeCoefficients);
```
Comment (AndrewMeadows) [inline libraries/avatars/src/HeadData.cpp]: Nice.

File: libraries/avatars/src/HeadData.h [hunk 2]
```diff
@@ -93,8 +93,8 @@ class HeadData {
     float _browAudioLift;
 
     QVector<float> _blendshapeCoefficients;
-    QVector<float> _baseBlendshapeCoefficients;
-    QVector<float> _currBlendShapeCoefficients;
+    QVector<float> _transientBlendshapeCoefficients;
+    QVector<float> _summedBlendshapeCoefficients;
```
Comment (AndrewMeadows) [inline libraries/avatars/src/HeadData.h]: Yay! Correct names are a Good Thing.

File: libraries/avatars/src/AvatarData.cpp [hunk 3]
```diff
@@ -445,17 +445,17 @@ QByteArray AvatarData::toByteArray(AvatarDataDetail dataDetail, quint64 lastSent
     if (hasFaceTrackerInfo) {
         auto startSection = destinationBuffer;
         auto faceTrackerInfo = reinterpret_cast<AvatarDataPacket::FaceTrackerInfo*>(destinationBuffer);
+        auto blendshapeCoefficients = _headData->getSummedBlendshapeCoefficients();
```
Comment (AndrewMeadows) [inline libraries/avatars/src/AvatarData.cpp]: For the other reviewers trying to follow all of the name changes, this is an important part of the fix: pack the **summed** coefficients into the byteArray.  The avatar-mixer and the recording pipeline use `AvatarData::toByteArray()` to obtain their blendshape data which should include script-set data (via `_blendshapeCoefficients`) and FaceTracker/voice-modulator data (via `_transientBlendshapeCoefficients`).

Suggestion, not a blocker: use `const auto&` here as a hint to the reader that the data should not change.  Rumor has it QVector<> performs automatic shared ref-counting on copies under the hood so this is more about readability than optimization.

**Comment by ZappoMan** [comment]
What exactly is this PR fixing?

**Comment by OmegaHeron** [comment]
Tested Windows 10/Ubuntu 16.04

Setup as follows;
Machine 1, Windows 10 running this PR Interface using my Mixamo AV.
Machine 2, Ubuntu 16.04 running this PR Interface and Stack using Photo Real Male AV.

Agent on machine 1 is able to see agent on machine 2 speak as expected.
Agent on machine 2 is able to see agent on machine 1 speak as expected.

Recorded agent on machine 1 speaking, being silent, speaking again and moving around.

Agent on machine 1 sees recorded agent as expected.
Agent on machine 2 sees recorded agent as expected.

In interest of being thorough -- it does seem the recording's eyes can do unexpected things like gazing at extreme right/left.  See similar in master so not this PR's problem, I think.

**Comment by ctrlaltdavid** [comment]
@ZappoMan This PR fixes avatars' mouths not moving when playing back a recording: https://highfidelity.fogbugz.com/f/cases/4632

**Pull request merged**